src/conversion/mapper.js

```diff
@@ -110,23 +110,43 @@ export default class Mapper {
 	/**
 	 * Unbinds given {@link module:engine/view/element~Element view element} from the map.
 	 *
+	 * **Note:** view-to-model binding will be removed, if it existed. However, corresponding model-to-view binding
+	 * will be removed only if model element is still bound to passed `viewElement`.
+	 *
+	 * This behavior lets for re-binding model element to another view element without fear of losing the new binding
+	 * when the previously bound view element is unbound.
+	 *
 	 * @param {module:engine/view/element~Element} viewElement View element to unbind.
 	 */
 	unbindViewElement( viewElement ) {
 		const modelElement = this.toModelElement( viewElement );
 
-		this._unbindElements( modelElement, viewElement );
+		this._viewToModelMapping.delete( viewElement );
+
+		if ( this._modelToViewMapping.get( modelElement ) == viewElement ) {
+			this._modelToViewMapping.delete( modelElement );
+		}
```

Comment:
Not sure isn't it simpler to just call both methods.

Reply:
It looks that it's not that easy after all. For example, take the default `remove` converter. You'd have to move the same logic there:

```javascript
for ( const child of ViewRange.createIn( removed ).getItems() ) {
	conversionApi.mapper.unbindViewElement( child );
}
```

The converter does not receive the removed item, just the remove position, remove length and a name of the removed item. So you can't just call `unbindModelElement` on it.